File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotContainRule.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import java.util.Locale;

import io.github.anderscheow.validator.rules.BaseRule;

public class NotContainRule extends BaseRule {

    private String keyword;

    public NotContainRule(@NonNull String keyword) {
        super(String.format(Locale.getDefault(), "Value does contain '%s'", keyword));
        this.keyword = keyword;
    }

    public NotContainRule(@NonNull String keyword, @StringRes int errorRes) {
        super(errorRes);
        this.keyword = keyword;
    }

    public NotContainRule(@NonNull String keyword, @NonNull String errorMessage) {
        super(errorMessage);
        this.keyword = keyword;
    }

    @Override
    public boolean validate(Object value) {
        if (value == null) {
            throw new NullPointerException();
        }

        if (value instanceof String) {
            return !((String) value).contains(keyword);
        }

        throw new ClassCastException("Required String value");
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/common/NotContainRule.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.Rule

class NotContainRule : Rule {

    private var keyword: String

    private var ignoreCase = false

    constructor(keyword: String, @StringRes errorRes: Int, ignoreCase: Boolean = false) :
            super(errorRes) {
        this.keyword = keyword
        this.ignoreCase = ignoreCase
    }

    constructor(keyword: String, errorMessage: String, ignoreCase: Boolean = false) :
            super(errorMessage) {
        this.keyword = keyword
        this.ignoreCase = ignoreCase
    }

    override fun validate(value: String?): Boolean {
        if (value == null) {
            throw NullPointerException()
        } else {
            return !value.contains(keyword, ignoreCase)
        }
    }
}

fun notContain(
    keyword: String,
    @StringRes errorRes: Int,
    ignoreCase: Boolean = false
): NotContainRule = NotContainRule(keyword, errorRes, ignoreCase)

fun notContain(
    keyword: String,
    errorMessage: String,
    ignoreCase: Boolean = false
): NotContainRule = NotContainRule(keyword, errorMessage, ignoreCase)